### DISCUSSION TITLE
Choosing what to restock

### DISCUSSION THREAD

**Discussion by TheDcol** [sunflower-land/sunflower-land#3773] [Ideas]
Hello, a small suggestion to allow players in choosing how many seeds and potentially tools they want to restock.

EG.

1 Restock = 1,000 points.

Kale = 10 Points
Corn = 5 Points

Upon restocking, i can choose 100 kale seeds = 1k points. 
Or a combination of 50 Kale seeds and 100 Corn seeds = 1k points.

Having a warehouse increases the points per restock (maybe to 1.2k)

Similar system would apply for Tools.

Issues:
- Would need to be balanced accordingly
- Chance of BB usage dropping dramatically?
- Player may restock wrong seeds, only to realise later they needed others for chores/others
- Seed limits will need to be changed (removed?)

Pros:
- Freedom of choice
- Diversity. Farms might focus on different types of crops. dedicated sellers?
- More kale per BB ;)

Would appreciate any feedback or further suggestions but the simpler the better

**Comment by drumdrumdrum3337** [comment]
What if, at least during the transitional period, both BB setups are available, but players need to flip the switch when restocking which setup they want to use, a dynamic one or a static one. This might mitigate transition pains, but also give us an idea what players actually prefer.

**Reply by TheDcol** [reply]
Sure, this could be an option. 'Default' restocking, then after x level we unlock a second option 'dynamic'

**Comment by COdefender17** [comment]
I love the intent behind this idea but I think ultimately this is a bad idea in reality. Imagine, you use a BB, you have points to get seeds, then you need to buy the seeds with coins, to plant, to harvest, to sell for SFL or ticket deliveries. One of the complaints about the crypto world is how difficult it is to navigate and all the layers of having to buy BTC to transfer to MM wallet to transfer to matic to get SFL or whatever the case may be. When I first joined web3 it was miserable. The last thing I want to do is have another layer to navigate as a new player. 

However, I do completely agree there needs to be some sort of love here. I would rather have a seed farmer where you can purchase certain bundles for a BB or somehow introduce your idea through an optional mechanic. So people could 1. use BB in its capacity now (even though the seed count is not adequate for the crop spot numbers in desert) or 2. use the BB at the seed farmer NPC at plaza to buy only 5 crop seeds but more quantity of those seeds. 

Another partial solution is increasing seed inventory limits so that if I wanted to play a full round of kale, I could just burn 2 BB while buying all the other seeds. This also allows players to stack up SF/potato seeds for a nice weekend power hour of planting 5000 seeds +. 

To me, the "issues" that need solved here are lack of seeds per crop considering 65 crop spots and the limit on seed inventory preventing certain style of game play. If people want to load up seed inventory, it just is locking up/staking coins basically which I don't see the harm.

**Comment by Polysemouse** [comment]
I believe this will benefit players greatly.  Currently the seed limit and fixed restock amount is a pain point.  By implementing stock points (obviously the scales need to be adjusted), it gives players the freedom on what they want to focus on their farms.

Ideally the stock points can be combined and used in both tools and seeds shop.  Players who want to focus on minerals can spend all their points on mining ores / fishing, and players who want to spam farming kale can devote all their points to buying kale seed.

Issues:

- Would need to be balanced accordingly
  - Yes.
- Chance of BB usage dropping dramatically?
  - Since players can specialize what they want, the restock amounts can potentally be decreased compared to what we have now.  Players can spend resources to upgrade their warehouse/toolshed to increase the stock points.
- Player may restock wrong seeds, only to realise later they needed others for chores/others
  - It is the player's fault 😆 They can always spend another BB and buy the seeds they want.  They still own the wrong seeds and can use them in other ways.
- Seed limits will need to be changed (removed?)
  - Seed limits should be removed.  Why tools have no limits while seeds have them?

**Comment by Dionis404** [comment]
As far as I understand, the point of the idea is to lower the cost of using the buck block? 

It's just that then players who have gold cauliflower would likely only plant cauliflower and sell it to goblins. Which could be bad for the economy as a whole. 
Right now, the seed refresh system forces the player to think about strategy.  
To make it easier for players to restock seeds and tools, you could lower the buck block price and make restock for seeds and tools separate from each other. 

UPD: 
Also, it should be taken into account that the owners of Foreman Beaver in the proposed system will receive more picks, which can also greatly affect the economy.

**Comment by TheDcol** [comment]
Rough version (so dont criticise too harshly pls :smile: )

- Normal 'Restock' option will show up until level x (20? 30?)
- After x level, when hitting 'Restock', a dialogue opens up giving you 2 options. 'Default' or 'Dynamic' (could be a better word). 
- If clicking 'Default', the usual amounts of seeds/tools are restocked.
- If clicking 'Dynamic', the player will be taken to a different screen able to choose what to restock and how much.
- Players can be given x many points, and seeds/tools will have points assigned to them. Player can choose how much they restock for what.
- We can set a 'minimum' amount for each seed/tool if we are worried about players simply restocking 1 item all the time. This would allow a bit of freedom in increasing stone/iron pickaxes but ending up with less rods, kale seeds etc.
